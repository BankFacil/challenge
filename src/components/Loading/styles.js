import styled from 'styled-components/native';
import {View} from 'react-native-animatable';

import theme from '../theme';

export const LoadingContainer = styled(View)`
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  z-index: 99;
  align-items: center;
  justify-content: center;
  background: ${theme.colors.light};
`;
