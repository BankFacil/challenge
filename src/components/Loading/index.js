import React, {memo, useState, useRef, useEffect} from 'react';
import {ActivityIndicator} from 'react-native';
import PropTypes from 'prop-types';

import {LoadingContainer} from './styles';
import theme from '../theme';

const Loading = ({show}) => {
  const [_show, _setShow] = useState();
  const view = useRef();

  useEffect(() => {
    if (_show !== show) {
      _setShow(show);
      if (show) {
        view.current.setNativeProps({zIndex: 99});
        view.current.fadeIn();
      } else {
        view.current
          .fadeOut()
          .then(() => view.current.setNativeProps({zIndex: -1}));
      }
    }
  }, [show]);

  return (
    <LoadingContainer ref={view}>
      <ActivityIndicator size="large" color={theme.colors.primary} />
    </LoadingContainer>
  );
};

Loading.propTypes = {
  show: PropTypes.bool.isRequired,
};

export default memo(Loading);
