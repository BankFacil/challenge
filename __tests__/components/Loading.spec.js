import React from 'react';
import renderer from 'react-test-renderer'

import Loading from '../../src/components/Loading';

describe('Loading', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = renderer.create(<Loading show={true} />).toJSON();
      expect(component).toMatchSnapshot();
    });

    it('should match to snapshot - hidden', () => {
      const component = renderer.create(<Loading show={false} />).toJSON();
      expect(component).toMatchSnapshot();
    });
  });
});
