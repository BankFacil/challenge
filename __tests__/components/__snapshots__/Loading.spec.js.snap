// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Rendering should match to snapshot - hidden 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "opacity": 1,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 99,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#3099DD"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`Loading Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 99,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#3099DD"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
